tests/testSchemaToMeta.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python
+
+# LSST Data Management System
+# Copyright 2015 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+
+"""
+This is a unittest for the SchemaToMeta class.
+
+@author  Jacek Becla, SLAC
+"""
+
+# standard library
+import logging as log
+import os
+import tempfile
+import unittest
+
+# useful for debugging
+# import pprint
+# pp = pprint.PrettyPrinter(indent=2)
+# pp.pprint(theTable)
+
+# local
+from lsst.metaserv.schemaToMeta import SchemaToMeta
+
+class TestS2M(unittest.TestCase):
+
+    def testBasics(self):
+        """
+        Basic test: load data for few tables.
+        """
+        (fd, fName) = tempfile.mkstemp()
+        theFile = os.fdopen(fd, "w")
+        theFile.write("""
+CREATE TABLE t1
+    -- <descr>This is t1 table.</descr>
+(
+    id int,
+        -- <descr>the t1.id</descr>
+    ra double,
+        -- <descr>right ascention</descr>
+        -- <ucd>pos.eq.ra</ucd>
+    decl double,
+        -- <ucd>pos.eq.dec</ucd>
+    s char,
+        -- <descr>the t1.s</descr>
+    v varchar(255),
+    PRIMARY KEY pk_t1_id (id),
+    INDEX idx_t1_s (s)
+) ENGINE=MyISAM;
+
+CREATE TABLE t2
+   -- <descr>This is
+   -- t2 table.</descr>
+(
+    id2 int,
+        -- <descr>the
+        -- t2.id2</descr>
+    s2 char,
+    v2 varchar(255),
+    UNIQUE (id2, s2)
+) ENGINE=InnoDB;
+""")
```

Comment:
First of all, I would test each non-trivial helper method individually to make sure it is matching the right range of inputs, including various spacings.
Then, for your overall test, you're missing coverage of many things:
* `DEFAULT`
* Three or more line descriptions for both tables and columns
* Units
* Empty end-description lines
* `NOT NULL`
* `FLOAT(0)` -> `FLOAT`